client/src/Components/Register/Register.js

```diff
@@ -0,0 +1,175 @@
+import React, { useState } from "react";
+import CustomButton from "../CustomButton/CustomButton";
+import FormInput from "../FormInput/FormInput";
+import { RegisterText } from "../../Constants/ConstantText";
+import "./Register.scss";
+import { userActions } from '../../_actions';
+import { connect } from "react-redux";
+
+const Register = (props) => {
+
+  const [formData, setFormData] = useState({
+    user: {
+      firstName: "",
+      lastName: "",
+      email: "",
+      password: "",
+      confirmPassword: ""
+    },
+    errors: {}
+  })
+
+  const handleChange = (event) => {
+    const { value, name } = event.target;
+    const { user } = formData;
+    setFormData({
+      user: {
+        ...user,
+        [name]: value
+      }
+    });
+  };
+
+  const validate = () => {
+    let errors = {};
+    let isValid = true;
+
+    if (!formData.user["firstName"]) {
+      isValid = false;
+      errors["firstName"] = "Please enter First Name.";
+    }
+
+    if (!formData.user["lastName"]) {
+      isValid = false;
+      errors["lastName"] = "Please enter Last Name.";
+    }
+
+    if (typeof formData.user["email"] !== "undefined") {
+      var pattern = new RegExp(/^(("[\w-\s]+")|([\w-]+(?:\.[\w-]+)*)|("[\w-\s]+")([\w-]+(?:\.[\w-]+)*))(@((?:[\w-]+\.)*\w[\w-]{0,66})\.([a-z]{2,6}(?:\.[a-z]{2})?)$)|(@\[?((25[0-5]\.|2[0-4][0-9]\.|1[0-9]{2}\.|[0-9]{1,2}\.))((25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\.){2}(25[0-5]|2[0-4][0-9]|1[0-9]{2}|[0-9]{1,2})\]?$)/i);
+      if (!pattern.test(formData.user["email"])) {
+        isValid = false;
+        errors["email"] = "Please enter valid email address.";
+      }
```

Comment:
extract validation logic to some external helper function. That function should only return true/false indicating success or failure

Reply:
done with changes